pkg/types/provider.go

```diff
@@ -2,6 +2,13 @@ package types
 
 import ocrtypes "github.com/smartcontractkit/libocr/offchainreporting2plus/types"
 
+const (
+	ProviderTypePlugin    = "plugin"
+	ProviderTypeMedian    = "median"
+	ProviderTypeFunctions = "functions"
+	ProviderTypeMercury   = "mercury"
```

Comment:
Maybe we should relocate this type? https://github.com/smartcontractkit/chainlink/blob/fbae1afbb38b71b6316460cefffd9ce58e94d354/core/services/job/models.go#L316
Used here: https://github.com/smartcontractkit/chainlink/blob/fbae1afbb38b71b6316460cefffd9ce58e94d354/core/services/job/models.go#L354

Reply:
Hmmm I think that makes sense.

Related: Why `PluginArgs.ProviderType` rather than `RelayerArgs.ProviderType`?

Reply:
> It feels to me like they're different, except that they happen to match at the moment.

Is there a chance that this distinction goes away completely before the symmetry is broken though? i.e. couldn't everything just configure an agnostic plugin provider in the future?

Reply:
>Related: Why PluginArgs.ProviderType rather than RelayerArgs.ProviderType?

I thought it should be PluginArgs since it's the plugin that's requiring a specific providerType. That being said, I looked at the doc comments for both types:
```// PluginArgs are the args required to create any OCR2 plugin components.
// It's possible that the plugin config might actually be different
// per relay type, so we pass the config directly through.```
And that makes me wonder if actually RelayArgs is the right place. Wdyt?

>Is there a chance that this distinction goes away completely before the symmetry is broken though? i.e. couldn't everything just configure an agnostic plugin provider in the future?

Yes that is the ultimate aim, but I think that's very far off since it would require removing product-specific providers altogether. Having a generic pluginType is much more proximate (i.e. should be implemented next quarter).